packages/phone-number-privacy/src/salt-generation/query-quota.ts

```diff
@@ -0,0 +1,56 @@
+import { AttestationsWrapper } from '@celo/contractkit/lib/wrappers/Attestations'
+import fetch from 'node-fetch'
+import config from '../config'
+import { getPerformedQueryCount } from '../database/wrappers/account'
+import { getContractKit } from '../web3/contracts'
+
+const NUM_ATTESTATIONS_REQUIRED = 3
+const contractKit = getContractKit()
+
+/*
+ * Returns how many queries the account can make based on the
+ * calculated query quota and the number of queries already performed.
+ */
+export async function getRemainingQueryCount(account: string, phoneNumber: string) {
+  const queryQuota = await getQueryQuota(account, phoneNumber)
+  const performedQueryCount = await getPerformedQueryCount(account).catch((reason) => {
+    // TODO [amyslawson] think of failure case here
+    console.error(reason)
+    return 0
+  })
+  return queryQuota - performedQueryCount
+}
+
+/*
+ * Calculates how many queries the caller has unlocked based on the algorithm
+ * unverifiedQueryCount + verifiedQueryCount + (queryPerTransaction * transactionCount)
+ */
+async function getQueryQuota(account: string, phoneNumber: string) {
+  let queryQuota: number = config.salt.unverifiedQueryMax
```

Comment:
nit, no need for explicit `: number` typing here right? TS should already know it